eng/native/build-commons.sh

```diff
@@ -163,7 +163,7 @@ build_native()
         popd
     else
         cmake_command=cmake
-        if [[ "$build_arch" == "wasm" ]]; then
+        if [[ "$build_arch" == "wasm" && "$__TargetOS" == "Browser" ]]; then
```

Comment:
should we be using `$targetOS` here? not sure what the difference is

```suggestion
        if [[ "$build_arch" == "wasm" && "$targetOS" == "Browser" ]]; then
```

Reply:
> should we be using `$targetOS` here? not sure what the difference is

I don't know either, but that broke stuff. Reverted.